src/sentry/static/sentry/app/views/eventsV2/eventDetails/transaction/eventBreakdown.tsx

```diff
@@ -0,0 +1,153 @@
+import React from 'react';
+import get from 'lodash/get';
+
+import {Event} from 'app/types';
+import {
+  SentryTransactionEvent,
+  SpanEntry,
+  SpanType,
+} from 'app/components/events/interfaces/spans/types';
+import {TraceContextType} from 'app/components/events/interfaces/spans/traceView';
+
+type EventBreakdownType = {
+  ops: {name: string; count: number; percentage: number; duration: number}[];
+  unknown: {count: number; percentage: number; duration: number} | undefined;
+};
+
+type Props = {
+  event: Event;
+};
+
+class EventBreakdown extends React.Component<Props> {
+  getTransactionEvent(): SentryTransactionEvent | undefined {
+    const {event} = this.props;
+
+    if (event.type === 'transaction') {
+      return event as SentryTransactionEvent;
+    }
+
+    return undefined;
+  }
+
+  generateCounts(): EventBreakdownType {
+    const event = this.getTransactionEvent();
+
+    if (!event) {
+      return {
+        ops: [],
+        unknown: undefined,
+      };
+    }
+
+    const spanEntry: SpanEntry | undefined = event.entries.find(
+      (entry: {type: string}) => entry.type === 'spans'
+    );
+
+    const spans: SpanType[] = get(spanEntry, 'data', []);
+
+    // track stats on spans with no operation
+    const spansWithNoOperation = {count: 0, duration: 0};
+
+    const numOfSpans = spans.length;
+
+    type AggregateType = {
+      [opname: string]: {
+        count: number;
+        duration: number; // num of seconds
+      };
+    };
+
+    const traceContext: TraceContextType | undefined = get(event, 'contexts.trace');
+    if (traceContext) {
+      spans.push({
+        op: traceContext.op,
+        timestamp: event.endTimestamp,
+        start_timestamp: event.startTimestamp,
+        trace_id: '',
+        span_id: '',
+        data: {},
+      });
+    }
+
+    const aggregateByOp: AggregateType = spans.reduce(
+      (aggregate: AggregateType, span: SpanType) => {
+        const op = span.op;
+        const duration = Math.abs(span.timestamp - span.start_timestamp);
+
+        if (typeof op !== 'string') {
+          spansWithNoOperation.count += 1;
+          spansWithNoOperation.duration += duration;
+
+          return aggregate;
+        }
+        const opStats = aggregate[op];
+
+        if (!opStats) {
+          aggregate[op] = {
+            count: 1,
+            duration,
+          };
+          return aggregate;
+        }
+
+        aggregate[op].count += 1;
+        aggregate[op].duration += duration;
+
+        return aggregate;
+      },
+      {}
+    );
+
+    const ops = Object.keys(aggregateByOp).map(opName => {
+      return {
+        name: opName,
+        count: aggregateByOp[opName].count,
+        percentage: aggregateByOp[opName].count / numOfSpans,
```

Comment:
Shouldn't the percentage be by time not number of spans?

Reply:
Oh damn. I did it incorrectly.